layers/ntp_test.go

```diff
@@ -0,0 +1,244 @@
+//******************************************************************************
+//
+// Copyright 2016 Ross N. Williams. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+//
+//******************************************************************************
+
+package layers
+
+import (
+	"github.com/google/gopacket"
+	"reflect"
+	"testing"
+)
+
+//******************************************************************************
+
+// check_ntp() uses the ntp.go code to analyse the packet bytes as an NTP UDP
+// packet and generate an NTP object. It then compares the generated NTP object
+// with the one provided and raises the alarm if there is any difference.
+// The desc argument is output with any failure message to identify the test.
+//
+func check_ntp(desc string, t *testing.T, packet_bytes []byte, p_expected_ntp *NTP) {
```

Comment:
func/var names camelCase, `checkNTP`, `packetBytes`, `expectedNTP` here and everywhere.